### PR TITLE
[bugfix-5506]: status in Instelling -> subcategory doesn't effect texts

### PR DESCRIPTION
**Pull request by SireeKoolenWijkstra**
Added useWatch to handle dependency of component on another component in StandardTextFiled

Because the parentcomponent Detail is not always rendered, the formValues are not always updated. React Forms has useWatch to check for changing of values in the form. FormValues are therefor removed and useWatch is implemented in CategoryForm where the showing of the standardTextFields is dependent on the status and the chosen standard text.

Ticket: [SIG-5506](https://gemeente-amsterdam.atlassian.net/browse/SIG-5506)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/signals/settings/categories/components/CategoryForm.tsx [hunk 1]
```diff
@@ -55,87 +53,123 @@ export interface Props {
...
-                    {...formMethods.register('name')}
-                    disabled={readOnly}
-                    hint="Het wijzigen van de naam heeft geen invloed op het type melding"
-                    id="name"
-                    label="Naam"
-                    name="name"
-                    readOnly={readOnly}
-                    type="text"
+}: Props) => {
+  const watchIs_Active = useWatch({
```
Comment (OscarBakker) [inline src/signals/settings/categories/components/CategoryForm.tsx:64]: Please change naming to camelcase

File: src/signals/settings/categories/components/CategoryForm.tsx [hunk 2]
```diff
@@ -55,87 +53,123 @@ export interface Props {
...
+                      name="name"
+                      readOnly={readOnly}
+                      type="text"
+                    />
+                  ) : (
+                    <>
+                      <StyledDefinitionTerm>
+                        <strong>Naam</strong>
+                      </StyledDefinitionTerm>
+                      <dd data-testid="name">{formMethods.getValues().name}</dd>
```
Comment (OscarBakker) [inline src/signals/settings/categories/components/CategoryForm.tsx:100]: formMethods.getValues('name') is more concise. Believe it should work

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx [hunk 3]
```diff
@@ -25,9 +25,10 @@ import { orderStandardTexts } from '../utils'
 type Props = {
   name: string
   onChange: (standardTexts: StandardText[]) => void
+  statusOption?: string
```
Comment (OscarBakker) [inline src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:28]: is a type statusOption

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx [hunk 4]
```diff
@@ -64,12 +65,17 @@ export const StandardTextsField = ({ name, onChange }: Props) => {
         (status) => status.value === selectedStatus
       )?.key
 
-      const orderedStandardTexts = data.results.filter(
+      let orderedStandardTexts = data.results.filter(
```
Comment (OscarBakker) [inline src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:68]: shorter version: 

const orderedStandardTexts = data.results
  .filter(result => result.state === state)
  .filter(item => !statusOption || item.active.toString() === statusOption)

**Review by OscarBakker — CHANGES_REQUESTED** [review]
It seems to work. couple of things

File: src/signals/settings/categories/components/CategoryForm.tsx (hunk 1 above)
Comment (SireeKoolenWijkstra) [reply src/signals/settings/categories/components/CategoryForm.tsx:64]: camelcased the shizz out of them.

File: src/signals/settings/categories/components/CategoryForm.tsx (hunk 2 above)
Comment (SireeKoolenWijkstra) [reply src/signals/settings/categories/components/CategoryForm.tsx:100]: all right, changed it.

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx (hunk 4 above)
Comment (SireeKoolenWijkstra) [reply src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:68]: changed it.

File: src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx (hunk 3 above)
Comment (SireeKoolenWijkstra) [reply src/signals/settings/categories/components/StandardTextsField/StandardTextsField.tsx:28]: all right, but I only need the key so I've set the prop to StatusOption['key]

**Review by OscarBakker — APPROVED** [review]

**Pull request merged**